scripts/Muon/GUI/Common/grouping_table_widget/grouping_table_widget_view.py

```diff
@@ -0,0 +1,307 @@
+
+    def _context_menu_add_group_action(self, slot):
+        add_group_action = QtGui.QAction('Add Group', self)
+        if len(self._get_selected_row_indices()) > 0:
+            add_group_action.setEnabled(False)
+        add_group_action.triggered.connect(slot)
+        return add_group_action
+
+    def _context_menu_remove_group_action(self, slot):
+        if len(self._get_selected_row_indices()) > 1:
+            # use plural if >1 item selected
+            remove_group_action = QtGui.QAction('Remove Groups', self)
+        else:
+            remove_group_action = QtGui.QAction('Remove Group', self)
+        if self.num_rows() == 0:
+            remove_group_action.setEnabled(False)
+        remove_group_action.triggered.connect(slot)
+        return remove_group_action
+
+    def _context_menu_add_pair_action(self, slot):
+        add_pair_action = QtGui.QAction('Add Pair', self)
+        if len(self._get_selected_row_indices()) != 2:
+            add_pair_action.setEnabled(False)
+        add_pair_action.triggered.connect(slot)
+        return add_pair_action
+
+    def contextMenuEvent(self, _event):
+        """Overridden method"""
+        self.menu = QtGui.QMenu(self)
+
+        self.add_group_action = self._context_menu_add_group_action(self.add_group_button.clicked.emit)
+        self.remove_group_action = self._context_menu_remove_group_action(self.remove_group_button.clicked.emit)
+        self.add_pair_action = self._context_menu_add_pair_action(self.add_pair_requested)
+
+        if self._disabled:
+            self.add_group_action.setEnabled(False)
+            self.remove_group_action.setEnabled(False)
+            self.add_pair_action.setEnabled(False)
+
+        self.menu.addAction(self.add_group_action)
+        self.menu.addAction(self.remove_group_action)
+        self.menu.addAction(self.add_pair_action)
+
+        self.menu.popup(QtGui.QCursor.pos())
+
+    # ------------------------------------------------------------------------------------------------------------------
+    # Slot connections
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def on_user_changes_group_name(self, slot):
+        self._validate_group_name_entry = slot
+
+    def on_user_changes_detector_IDs(self, slot):
+        self._validate_detector_ID_entry = slot
+
+    def on_add_group_button_clicked(self, slot):
+        self.add_group_button.clicked.connect(slot)
+
+    def on_remove_group_button_clicked(self, slot):
+        self.remove_group_button.clicked.connect(slot)
+
+    def on_table_data_changed(self, slot):
+        self._on_table_data_changed = slot
+
+    def add_pair_requested(self):
+        selected_names = self.get_selected_group_names()
+        self.addPairRequested.emit(selected_names[0], selected_names[1])
+
+    def on_item_changed(self):
```

Comment:
what is the purpose of this?

Reply:
Currently this does nothing. We may in the future want to take some action when an item is changed but in the short term probably easier to remove and then add back in if needed